feat(axum): provide docs for Axum

lcian opened the pull request:
<!-- Use this checklist to make sure your PR is ready for merge. You may delete any sections you don't need. -->

## DESCRIBE YOUR PR
Provides getting started docs for Axum.
The structure is very similar to the one used in https://github.com/getsentry/sentry-docs/pull/13821, only the code snippets are different.

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): <!-- ENTER DATE HERE -->
- [ ] Other deadline: <!-- ENTER DATE HERE -->
- [X] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## LEGAL BOILERPLATE

<!-- Sentry employees and contractors can delete or ignore this section. -->

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

lcian commented:
> @lcian Looks like the build deploys were canceled. I can't seem to try to deploy or redeploy them. Can you? Or can you force push to this branch so we can have those builds run?

Now it's working!

sfanahata commented on docs/platforms/rust/guides/axum/index.mdx:
```diff
@@ -0,0 +1,99 @@
+description: "Learn about monitoring your axum application with Sentry."
+---
+
+The Sentry SDK offers a middleware for the [`axum`](https://github.com/tokio-rs/axum) framework that supports:
```
```suggestion
The Sentry SDK offers a middleware for the [axum](https://github.com/tokio-rs/axum) framework that supports:
```
This is looking a bit strange as code and a link. I suggest just keeping it as a hyperlink.

sfanahata commented on docs/platforms/rust/guides/axum/index.mdx:
```diff
@@ -0,0 +1,99 @@
+
+</Alert>
+
+To view and resolve the recorded error, log into [sentry.io](https://sentry.io) and select your project. Clicking on the error's title will open a page where you can see detailed information and mark it as resolved.
```
Would that error now be found under `issues > errors & outages` in our new nav? It would be helpful to be specific about where to find the error to select.

sfanahata reviewed (commented):
@lcian - a few things should be cleaned up before we merge. It's close though! Let me know if you need any help.

lcian commented:
Thanks for the helpful feedback @sfanahata 
I've addressed it and also added the `Tracing` checkbox that we're using across SDKs (disabled by default) to turn on and off the corresponding code.

lcian commented:
Using a normal link because the PlatformLink was causing a 404
